controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -155,6 +155,11 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 		return ctrl.Result{}, err
 	}
 
+	err = r.setPolicyDesiredStateDefaults(instance)
```

Comment:
Why are you updating the policy instance and not the enactment directly?
For example. you had to change the `incrementUnavailableNodeCount` behaviour because of the policy instance change, updating the enactment directly would avoid it.

Reply:
I have considered two options here:
1. update the enactment + apply the defaults to the policy right before setting the desiredState in `pkg/helper/client.go ApplyDesiredState`
2. just update the NNCP, initialize the NNCE based on the updated version and use it to also set the desired state

I considered the 2. option to be more straightforward, because it doesn't mean repeating the same work - parsing the desiredState and applying defaults to it.
It did require to change the `incrementUnavailableNodeCount` so the the function doesn't modify the parameter passed to it, which I think is better anyway.

Reply:
What about option 3 - use the desired state from the enactment on `ApplyDesiredState`. I believe it is more correct even not related to this PR. Since the enactment represent the desired state of the current node.ApplyDesiredState
And this way you won't have to carry about updating the policy instance.

Reply:
Yes, that makes sense as well. Updated accordingly